bin/oneoff/backfill_data/backfill_level_references.rb

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env ruby
+
+# https://codedotorg.atlassian.net/browse/TEACH-935
+# Context
+# Reference guides for curriculum was migrated to a framework that supports versioning
+# But, the corresponding change needed to update version for linked reference guides when
+# cloning scripts is currently missing. That is tracked by task TEACH-
+
+# This script is a backfill to fix the reference guide version for scripts that are already
+# cloned for 2024.
+
+require_relative '../../../dashboard/config/environment'
+
+# Override for the dry run variable using an commandline argument
+DRY_RUN = !((ARGV.find {|arg| arg.casecmp('-dryrun')}).nil?)
+
+def update_reference_guide_format(unit_group_name, course_version_id, reference_guide_link)
+  # Reference guide formats
+  # in the old unversioned framework - /docs/concepts/patterns/random-list-access/
+  # in the versioned framework - /courses/csd-2024/guides/animation-tab
+  if !reference_guide_link.starts_with?("/docs/") && !reference_guide_link.starts_with?("/courses/")
+    return reference_guide_link
+  end
+
+  reference_guide_key = reference_guide_link.split("/").reject(&:blank?).map(&:strip).last
+
+  reference_guide = ReferenceGuide.find_by(key: reference_guide_key, course_version_id: course_version_id)
+  if reference_guide.nil?
+    ref_guide_missing_err = "Unable to find reference guide with key #{reference_guide_key}"
+    puts ref_guide_missing_err
+
+    raise ref_guide_missing_err unless DRY_RUN
```

Comment:
During dry run locally, there were only two instances where the versioned reference guide didn't follow this format. So just log the error instead of erroring out of the script as these can be manually fixed later.